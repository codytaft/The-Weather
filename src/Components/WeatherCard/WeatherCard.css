.weather-card {
  flex-wrap: wrap;
  margin: 1rem;
  border: 0.1rem solid whitesmoke;
  width: 17rem;
  text-align: center;
  border-radius: 0.2rem;
  color: whitesmoke;
}
