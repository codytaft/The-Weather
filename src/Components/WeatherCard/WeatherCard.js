import React, { Component } from 'react';
import './WeatherCard.css';
class WeatherCard extends Component {
  render() {
    return (
      <div className="weather-card">
        <h1>{this.props.location}</h1>
        <h2>{this.props.forecast.dt_txt}</h2>
        <h2>{Math.round(this.props.forecast.main.temp)}Ëš</h2>
      </div>
    );
  }
}

export default WeatherCard;
