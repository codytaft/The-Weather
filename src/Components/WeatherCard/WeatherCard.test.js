import React from 'react';
import { shallow } from 'enzyme';
import WeatherCard from './WeatherCard';

describe('WeatherCard', () => {
  let wrapper;
  let mockLocation = 'corsicana';
  let mockForecast = {
    main: '65'
  };

  beforeEach(() => {
    wrapper = shallow(
      <WeatherCard location={mockLocation} forecast={mockForecast} />
    );
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot if location is a city', () => {
    let mockLocation = 'corsicana';
    wrapper = shallow(
      <WeatherCard location={mockLocation} forecast={mockForecast} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot if location is a zip code', () => {
    let mockLocation = '75110';
    wrapper = shallow(
      <WeatherCard location={mockLocation} forecast={mockForecast} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
