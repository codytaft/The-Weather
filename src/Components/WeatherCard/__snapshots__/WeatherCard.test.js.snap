// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherCard should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WeatherCard
    forecast={
      Object {
        "main": "65",
      }
    }
    location="corsicana"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          corsicana
        </h1>,
        <h2 />,
        <h2>
          NaN
          ˚
        </h2>,
      ],
      "className": "weather-card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "corsicana",
        },
        "ref": null,
        "rendered": "corsicana",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            NaN,
            "˚",
          ],
        },
        "ref": null,
        "rendered": Array [
          NaN,
          "˚",
        ],
        "type": "h2",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            corsicana
          </h1>,
          <h2 />,
          <h2>
            NaN
            ˚
          </h2>,
        ],
        "className": "weather-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "corsicana",
          },
          "ref": null,
          "rendered": "corsicana",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              NaN,
              "˚",
            ],
          },
          "ref": null,
          "rendered": Array [
            NaN,
            "˚",
          ],
          "type": "h2",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;

exports[`WeatherCard should match the snapshot if location is a city 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WeatherCard
    forecast={
      Object {
        "main": "65",
      }
    }
    location="corsicana"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          corsicana
        </h1>,
        <h2 />,
        <h2>
          NaN
          ˚
        </h2>,
      ],
      "className": "weather-card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "corsicana",
        },
        "ref": null,
        "rendered": "corsicana",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            NaN,
            "˚",
          ],
        },
        "ref": null,
        "rendered": Array [
          NaN,
          "˚",
        ],
        "type": "h2",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            corsicana
          </h1>,
          <h2 />,
          <h2>
            NaN
            ˚
          </h2>,
        ],
        "className": "weather-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "corsicana",
          },
          "ref": null,
          "rendered": "corsicana",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              NaN,
              "˚",
            ],
          },
          "ref": null,
          "rendered": Array [
            NaN,
            "˚",
          ],
          "type": "h2",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;

exports[`WeatherCard should match the snapshot if location is a zip code 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WeatherCard
    forecast={
      Object {
        "main": "65",
      }
    }
    location="75110"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          75110
        </h1>,
        <h2 />,
        <h2>
          NaN
          ˚
        </h2>,
      ],
      "className": "weather-card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "75110",
        },
        "ref": null,
        "rendered": "75110",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            NaN,
            "˚",
          ],
        },
        "ref": null,
        "rendered": Array [
          NaN,
          "˚",
        ],
        "type": "h2",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            75110
          </h1>,
          <h2 />,
          <h2>
            NaN
            ˚
          </h2>,
        ],
        "className": "weather-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "75110",
          },
          "ref": null,
          "rendered": "75110",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              NaN,
              "˚",
            ],
          },
          "ref": null,
          "rendered": Array [
            NaN,
            "˚",
          ],
          "type": "h2",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
